projects/plugins/jetpack/class.jetpack-plan.php

```diff
@@ -344,3 +344,44 @@ public static function supports( $feature ) {
 		return false;
 	}
 }
+
+/**
+ * Modify the allowed mimes types when uploading files,
+ * acoording to Jetpack module and/or feature.
+ *
+ * @param array Mime types provided by the `upload_mimes` filter.
+ * @return array Filtered mime types array.
+ */
+function jetpack_filter_mimetypes_by_modules_and_features( $mimes ) {
+	// Not connected. Bail early.
+	if ( ! Jetpack::is_connection_ready() ) {
+		return $mimes;
+	}
+
+	// `videopress` module is not active, Bail early.
+	if ( Jetpack::is_module_active( 'videopress' ) ) {
+		return $mimes;
+	};
+
+	// `videopress` feature is supported. Bail early.
+	if ( Jetpack_Plan::supports( 'videopress' ) ) {
+		return $mimes;
+	};
+
+
+	// Remove `video/` mimes from the list.
+	return array_filter(
```

Comment:
I'm not sure we want to remove the ability to upload videos. If VideoPress is not available/active then people should still be able to upload videos to their site.

Reply:
agree, something (wrongly) handled in the block above. When the module is not active, bail early.

Reply:
fixed here 05912df5b6cd48a2e7b41e76fb675452bc19a93c

Reply:
Under what circumstances would we get here? If the module is active then the plan must support it, and if it's inactive then we bail early.

I don't think this class is available on WPCOM, and it's only there we want to remove these mime types. It's possible I'm just confused. There are a lot of moving parts to this.